core/src/test/java/io/questdb/test/cutlass/pgwire/PGWireMetricsQueryCounterTest.java

```diff
@@ -0,0 +1,580 @@
+            }
+        });
+    }
+
+    @Test
+    public void testQueryCountForSimpleStatement() throws Exception {
+        // former test name testShowTableInBlock
+        assertMemoryLeak(() -> {
+            try (PGWireMetricsQueryCounterTest.PGTestSetup test = new PGWireMetricsQueryCounterTest.PGTestSetup()) {
+                Statement statement = test.statement;
+
+                boolean hasResult = statement.execute(
+                        "SHOW TABLES; SELECT '15';");
+                assertResults(statement, hasResult, data(row(configuration.getSystemTableNamePrefix() + "text_import_log")), data(row(15L)));
+
+                long startedQueriesForTest = metrics.pgWire().startedQueriesCounter().getValue() - totalQueriesStarted;
+                long completedQueriesForTest = metrics.pgWire().completedQueriesCounter().getValue() - totalQueriesCompleted;
+                Assert.assertEquals(4, startedQueriesForTest);
+                Assert.assertEquals(4, completedQueriesForTest);
+            }
+        });
+    }
+
+    @Test
+    public void testQueryCountForSingleBlockStatementWithSeveralQueries() throws Exception {
+        // former test name testRunSeveralQueriesInSingleBlockStatementReturnsAllSelectResultsInOrder
+        assertMemoryLeak(() -> {
+            try (PGWireMetricsQueryCounterTest.PGTestSetup test = new PGWireMetricsQueryCounterTest.PGTestSetup()) {
+
+                boolean hasResult = test.statement.execute(
+                        "create table test(l long, s string);" +
+                                "insert into test values(1, 'a');" +
+                                "insert into test values(2, 'b');" +
+                                "select * from test;");
+
+                assertResults(test.statement, hasResult, PGMultiStatementMessageTest.Result.ZERO, count(1), count(1),
+                        data(row(1L, "a"), row(2L, "b")));
+
+                long startedQueriesForTest = metrics.pgWire().startedQueriesCounter().getValue() - totalQueriesStarted;
+                long completedQueriesForTest = metrics.pgWire().completedQueriesCounter().getValue() - totalQueriesCompleted;
+                Assert.assertEquals(6, startedQueriesForTest);
+                Assert.assertEquals(6, completedQueriesForTest);
+            }
+        });
+    }
+
+    @Test
+    public void testQueryCountForSmallQuery() throws Exception {
+        // former test name testCreateInsertAsSelectReturnsProperUpdateCount
+        assertMemoryLeak(() -> {
+            try (PGWireMetricsQueryCounterTest.PGTestSetup test = new PGWireMetricsQueryCounterTest.PGTestSetup()) {
+                Statement statement = test.statement;
+
+                boolean hasResult = statement.execute("CREATE TABLE test(l long, s string); " +
+                        "INSERT INTO test select x,'str' from long_sequence(20);");
+                assertResults(statement, hasResult, PGMultiStatementMessageTest.Result.ZERO, count(20));
+
```

Comment:
It would be better to hide the subtraction logic in a method to make things clearer (and subtract 2 inside) .